src/invoke-store.ts

```diff
@@ -15,112 +9,160 @@ const PROTECTED_KEYS = {
     }
 
-    const context = this.storage.getStore();
-    if (context) {
-      context[key] = value;
+    this.currentContext = this.currentContext || {};
+    this.currentContext[key] = value;
+  }
+
+  run<T>(context: Context, fn: () => T): T {
+    this.currentContext = context;
+    try {
+      return fn();
+    } finally {
+      this.currentContext = undefined;
     }
   }
 
-  /**
-   * Get the current request ID
-   */
-  public static getRequestId(): string {
-    return this.get<string>(this.PROTECTED_KEYS.REQUEST_ID) ?? "-";
+  getInstance(): InvokeStoreBase {
+    return this;
+  }
+}
+
+// Multi Context Implementation
+class InvokeStoreMulti extends InvokeStoreBase {
+  private als!: import("node:async_hooks").AsyncLocalStorage<Context>;
+
+  constructor() {
+    super();
   }
 
-  /**
-   * Get the current X-ray trace ID
-   */
-  public static getXRayTraceId(): string | undefined {
-    return this.get<string>(this.PROTECTED_KEYS.X_RAY_TRACE_ID);
+  static async create(): Promise<InvokeStoreMulti> {
+    const instance = new InvokeStoreMulti();
+    const asyncHooks = await import('node:async_hooks');
+    instance.als = new asyncHooks.AsyncLocalStorage<Context>();
+    return instance;
   }
 
-  /**
-   * Get the current tenant ID
-   */
-  public static getTenantId(): string | undefined {
-    return this.get<string>(this.PROTECTED_KEYS.TENANT_ID);
+  getContext(): Context | undefined {
+    return this.als.getStore();
   }
 
-  /**
-   * Check if we're currently within an invoke context
-   */
-  public static hasContext(): boolean {
-    return this.storage.getStore() !== undefined;
+  hasContext(): boolean {
+    return this.als.getStore() !== undefined;
   }
 
-  /**
-   * Check if a key is protected (readonly Lambda context field)
-   */
-  private static isProtectedKey(key: string | symbol): boolean {
-    return (
-      key === this.PROTECTED_KEYS.REQUEST_ID ||
-      key === this.PROTECTED_KEYS.X_RAY_TRACE_ID
-    );
+  get<T = unknown>(key: string | symbol): T | undefined {
+    return this.als.getStore()?.[key] as T | undefined;
   }
-}
 
-let instance: typeof InvokeStoreImpl;
+  set<T = unknown>(key: string | symbol, value: T): void {
+    if (this.isProtectedKey(key)) {
+      throw new Error(`Cannot modify protected Lambda context field: ${String(key)}`);
+    }
 
-if (!noGlobalAwsLambda && globalThis.awslambda?.InvokeStore) {
-  instance = globalThis.awslambda.InvokeStore;
-} else {
-  instance = InvokeStoreImpl;
+    const store = this.als.getStore();
+    if (!store) {
+      throw new Error('No context available');
+    }
+    
+    store[key] = value;
+  }
 
-  if (!noGlobalAwsLambda && globalThis.awslambda) {
-    globalThis.awslambda.InvokeStore = instance;
+  run<T>(context: Context, fn: () => T): T {
+    return this.als.run(context, fn);
   }
 }
 
-export const InvokeStore = instance;
+interface InvokeStoreConfig {
+  env?: NodeJS.ProcessEnv;
+}
+
+export namespace InvokeStore {
```

Comment:
Mark it as public and add description

```suggestion
/**
 * ToDo: add description
 * @public
 */
export namespace InvokeStore {
```